src/v/config/validators.cc

```diff
@@ -127,4 +127,22 @@ std::optional<ss::sstring> validate_0_to_1_ratio(const double d) {
     return std::nullopt;
 }
 
+std::optional<ss::sstring>
+validate_non_empty_string_vec(const std::vector<ss::sstring>& vs) {
+    for (const auto& s : vs) {
+        if (s.empty()) {
+            return "Empty strings are not valid in this collection";
+        }
+    }
+    return std::nullopt;
+}
+
+std::optional<ss::sstring>
+validate_non_empty_string_opt(const std::optional<ss::sstring>& os) {
+    if (os.has_value() && os.value().empty()) {
```

Comment:
not even a nit, but this can be just os == "" (if ss::sstring::operator== takes const char* parameter directly) https://gcc.godbolt.org/z/TaYGr5ePo